pandera/engines/pyspark_engine.py

```diff
@@ -178,7 +179,7 @@ class String(DataType, dtypes.String):  # type: ignore
 
 
 @Engine.register_dtype(
-    equivalents=[int, "int", "IntegerType()", pst.IntegerType(), pst.IntegerType],  # type: ignore
+    equivalents=[int, "int", "IntegerType", "IntegerType()", pst.IntegerType(), pst.IntegerType],  # type: ignore
```

Comment:
btw the "IntegerType()" string equivalent doesn't really make sense, since this is just a string alias the `()` doesn't actually do anything. I think the string alias "IntegerType" is more appropriate, same for the rest of the types.